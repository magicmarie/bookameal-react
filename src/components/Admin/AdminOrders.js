import React, { Component } from "react";
import Loader from "react-loader";
import { notify } from "react-notify-toast";
import axiosInstance from "../common/Apicalls";
import Pagination from "../common/Pagination";

const Order = props => (
  <div className="menu-item">
    <div className="row">
      <div className="col-md-4">{props.mealName}</div>
      <div className="price col-md-2">{props.price}</div>
      <div className="col-md-3 text-center">{props.customerName}</div>
      <div className="col-sm-3 text-center">
        <select>
          <option value="Delivered">Delivered</option>
          <option value="Not delivered">Not delivered</option>
          <option value="In progress">In progress</option>
        </select>
      </div>
    </div>
  </div>
);

class AdminDashboard extends Component {
  state = {
    total: null,
    orders: [],
    loaded: false,
    nextPage: null,
    previousPage: null,
    currentPage: null,
    pages: null,
    totalCount: null
  };
  //get all orders from admin's meals
  getUserOrders = () => {
    const page = this.state.currentPage || 1;
    axiosInstance
      .get("/orders", { params: { page } })
      .then(response => {
        const {
          total,
          currentPage,
          orders,
          nextPage,
          pages,
          perPage,
          previousPage,
          totalCount
        } = response.data.order_items;
        this.setState({
          total,
          currentPage,
          orders,
          nextPage,
          pages,
          perPage,
          previousPage,
          totalCount,
          loaded: true
        });
      })
      .catch(error => {
        if (error.response) {
          const { status } = error.response;
          if (status === 404) {
            this.setState({
              orders: []
            });
          } else if (status === 401) {
            localStorage.removeItem("token");
          }
        } else if (error.request) {
          notify.show("Wrong request", "warning", 2500);
        }
      });
  };
  //pagination
  changePage = selectedPage => {
    this.setState(
      {
        currentPage: selectedPage
      },
      () => this.getUserOrders()
    );
  };

  // always called before the component's first render
  // fetches the orders list
  componentWillMount() {
    this.getUserOrders();
  }
  render() {
    const {
      total,
      orders,
      loaded,
      currentPage,
      pages,
      nextPage,
      previousPage
    } = this.state;

    const ordersDetails =
      orders.length === 0 ? (
        <div>No orders Found</div>
      ) : (
        orders.map(order => (
          <div key={order.id}>
            <Order
              id={order.id}
              mealName={order.meal_name}
              price={order.price}
              customerName={order.customerName}
            />
          </div>
        ))
      );

    return (
      <div className="container-fluid">
        <h2 className="header text-center">Orders</h2>
        <div className="card mb-2" style={{ width: "10rem" }}>
          <div className="card-body">
            <h5 className="card-title">Total amount</h5>
            <p className="card-text">
              {total}
              UGX
            </p>
          </div>
        </div>
        <Loader loaded={loaded}>
          <div>{ordersDetails}</div>
        </Loader>
        <Pagination
          currentPage={currentPage}
          previousPage={previousPage}
          nextPage={nextPage}
          pages={pages}
          changePage={this.changePage}
        />
      </div>
    );
  }
}
export default AdminDashboard;
