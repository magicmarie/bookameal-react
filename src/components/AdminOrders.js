import React, { Component } from "react";
import Loader from "react-loader";
import { notify } from "react-notify-toast";
import axiosInstance from "./common/Apicalls";

const Order = props => (
  <div className="menu-item">
    <div className="row">
      <div className="col-md-4">{props.mealName}</div>
      <div className="price col-md-2">{props.price}</div>
      <div className="col-md-3">{props.customerName}</div>
      <div className="col-sm-3 text-center">
        <select>
          <option value="Delivered">Delivered</option>
          <option value="Not delivered">Not delivered</option>
          <option value="In progress">In progress</option>
        </select>
      </div>
    </div>
  </div>
);

class AdminDashboard extends Component {
  state = {
    orders: [],
    loaded: false
  };
  //get all orders from admin's meals
  getUserOrders = () => {
    axiosInstance
      .get("/orders")
      .then(response => {
        const orders = response.data.order_items.orders;
        this.setState({ orders, loaded: true });
      })
      .catch(error => {
        if (error.response) {
          const { status } = error.response;
          if (status === 404) {
            this.setState({
              orders: []
            });
          } else if (status === 401) {
            localStorage.removeItem("token");
          }
        } else if (error.request) {
          notify.show("Wrong request", "warning", 2500);
        }
      });
  };

  // always called before the component's first render
  // fetches the orders list
  componentWillMount() {
    this.getUserOrders();
  }
  render() {
    const { orders, loaded } = this.state;

    const ordersDetails =
      orders.length === 0 ? (
        <div>No orders Found</div>
      ) : (
        orders.map(order => (
          <div key={order.id}>
            <Order
              id={order.id}
              mealName={order.meal_name}
              price={order.price}
              customerName={order.customerName}
            />
          </div>
        ))
      );

    return (
      <div className="container-fluid">
        <h2 className="header text-center">Orders</h2>
        <Loader loaded={loaded}>
          <div>{ordersDetails}</div>
        </Loader>
      </div>
    );
  }
}
export default AdminDashboard;
