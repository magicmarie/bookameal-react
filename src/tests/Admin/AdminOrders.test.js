import React from "react";
import { shallow } from "enzyme";
import MockAdapter from "axios-mock-adapter";
import AdminOrders from "../../components/Admin/AdminOrders";
import axiosInstance from "../../components/common/Apicalls";

describe("AdminMenu component", () => {
  let data;
  let wrapper;
  beforeEach(() => {
    data = {
      currentPage: 1,
      orders: [{ id: 1, meal_name: "test", price: 1000 }],
      nextPage: 2,
      pages: 2,
      previousPage: ""
    };
    wrapper = shallow(<AdminOrders />);
  });
  it("renders successfully", async () => {
    const axiosMock = new MockAdapter(axiosInstance);
    await axiosMock.onGet("/orders").reply(200, {
      order_items: data
    });
    wrapper.instance().setState({ menus: data });
  });
  it("renders successfully", () => {
    const axiosMock = new MockAdapter(axiosInstance);
    axiosMock.onGet("/orders").reply(401, {
      order_items: data
    });
  });
  it("changes page", () => {
    const { currentPage } = wrapper.instance().state;
    expect(currentPage).toBe(null);
    wrapper.instance().changePage(1);
    expect(wrapper.instance().state.currentPage).toBe(1);
  });
  it("renders successfully", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
