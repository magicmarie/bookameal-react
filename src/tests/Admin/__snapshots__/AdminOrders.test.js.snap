// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminMenu component renders successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminDashboard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="header text-center"
>
          Orders
</h2>,
        <div
          className="card mb-2"
          style={
                    Object {
                              "width": "10rem",
                            }
          }
>
          <div
                    className="card-body"
          >
                    <h5
                              className="card-title"
                    >
                              Total amount
                    </h5>
                    <p
                              className="card-text"
                    >
                              UGX
                    </p>
          </div>
</div>,
        <Unknown
          component="div"
          loaded={false}
          loadedClassName="loadedContent"
          parentClassName="loader"
>
          <div>
                    <div>
                              No orders Found
                    </div>
          </div>
</Unknown>,
        <Pagination
          changePage={[Function]}
          currentPage={null}
          nextPage={null}
          pages={null}
          previousPage={null}
/>,
      ],
      "className": "container-fluid",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Orders",
          "className": "header text-center",
        },
        "ref": null,
        "rendered": "Orders",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="card-body"
>
            <h5
                        className="card-title"
            >
                        Total amount
            </h5>
            <p
                        className="card-text"
            >
                        UGX
            </p>
</div>,
          "className": "card mb-2",
          "style": Object {
            "width": "10rem",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5
                className="card-title"
>
                Total amount
</h5>,
              <p
                className="card-text"
>
                UGX
</p>,
            ],
            "className": "card-body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Total amount",
                "className": "card-title",
              },
              "ref": null,
              "rendered": "Total amount",
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  undefined,
                  "UGX",
                ],
                "className": "card-text",
              },
              "ref": null,
              "rendered": Array [
                undefined,
                "UGX",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div>
            <div>
                        No orders Found
            </div>
</div>,
          "component": "div",
          "loaded": false,
          "loadedClassName": "loadedContent",
          "parentClassName": "loader",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              No orders Found
</div>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No orders Found",
            },
            "ref": null,
            "rendered": "No orders Found",
            "type": "div",
          },
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "changePage": [Function],
          "currentPage": null,
          "nextPage": null,
          "pages": null,
          "previousPage": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="header text-center"
>
            Orders
</h2>,
          <div
            className="card mb-2"
            style={
                        Object {
                                    "width": "10rem",
                                  }
            }
>
            <div
                        className="card-body"
            >
                        <h5
                                    className="card-title"
                        >
                                    Total amount
                        </h5>
                        <p
                                    className="card-text"
                        >
                                    UGX
                        </p>
            </div>
</div>,
          <Unknown
            component="div"
            loaded={false}
            loadedClassName="loadedContent"
            parentClassName="loader"
>
            <div>
                        <div>
                                    No orders Found
                        </div>
            </div>
</Unknown>,
          <Pagination
            changePage={[Function]}
            currentPage={null}
            nextPage={null}
            pages={null}
            previousPage={null}
/>,
        ],
        "className": "container-fluid",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Orders",
            "className": "header text-center",
          },
          "ref": null,
          "rendered": "Orders",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="card-body"
>
              <h5
                            className="card-title"
              >
                            Total amount
              </h5>
              <p
                            className="card-text"
              >
                            UGX
              </p>
</div>,
            "className": "card mb-2",
            "style": Object {
              "width": "10rem",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h5
                  className="card-title"
>
                  Total amount
</h5>,
                <p
                  className="card-text"
>
                  UGX
</p>,
              ],
              "className": "card-body",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Total amount",
                  "className": "card-title",
                },
                "ref": null,
                "rendered": "Total amount",
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    "UGX",
                  ],
                  "className": "card-text",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  "UGX",
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div>
              <div>
                            No orders Found
              </div>
</div>,
            "component": "div",
            "loaded": false,
            "loadedClassName": "loadedContent",
            "parentClassName": "loader",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div>
                No orders Found
</div>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No orders Found",
              },
              "ref": null,
              "rendered": "No orders Found",
              "type": "div",
            },
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "changePage": [Function],
            "currentPage": null,
            "nextPage": null,
            "pages": null,
            "previousPage": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
